### ISSUE TITLE
ValueError : Sample Larger than population or is negative for 5 shot 2 way problem

### ISSUE THREAD

**Issue by haricash** [sicara/easy-few-shot-learning#68]
**Problem**
I am new to FSL and have a simple problem in my scientific domain that I thought I would try as a learning example. I am trying to perform classical training for a 5 shot 2 way problem. When I am running the code from the tutorial notebook as it is after using EasySet to create a custom data object, I am getting the following error when I encounter the validation epoch during my training: 

`ValueError : Sample Larger than population or is negative
`

**Considered solutions**
I've tried changing the batch size and n_workers so far, and neither have worked

**How can we help**
I can't figure out what is going wrong here. I am very new to machine learning and would love to have your help in any way possible!

**Comment by diego91964** [comment]
Based on documentation of the 'TaskSampler" you need attempt to the number of images in yout dataset.

n_way: number of classes in one task
n_shot: number of support images for each class in one task
n_query: number of query images for each class in one task

How many images by class in your dataset ? Its necessary more than n_shot + n_query images by class, and you need be attempt wich  small number of images results in a overfitting problem.

**Comment by ebennequin** [comment]
It's the 3rd time this issue is risen after #44 and #45. I'm labeling this as enhancement: we need a clear warning message in `TaskSampler`'s init when the dataset's population doesn't fit the shape of the expected few-shot tasks.

**Comment by haricash** [comment]
Thanks for your help @diego91964 ! I realized where I was going wrong. I had misunderstood the purpose of the query set, and had set the number of query items as the number of samples in my test set. I went through the docs once more and realised my mistake! Thanks a lot.

**Comment by haricash** [comment]
@ebennequin will you just be requiring to raise a warning message? I could do this as a first contribution if that's the case.

**Comment by ebennequin** [comment]
@haricash that would be really helpful, thank you!

Indeed, I think we just need to check that we have more than `n_way` classes in the dataset, and more than `n_shot + n_query` items for each class, and raise an explicit error message in each situation.

**Comment by shraddha291996** [comment]
> @haricash that would be really helpful, thank you!
> 
> Indeed, I think we just need to check that we have more than `n_way` classes in the dataset, and more than `n_shot + n_query` items for each class, and raise an explicit error message in each situation.

Hi,
I was facing similar issue, but now it is resolved as above suggestions. Although I was wondering why it is necessary to have more classes in the dataset than n_way? What happens with the additional ones? I hope my question is clear. 
Thank you in advance!

**Comment by ebennequin** [comment]
You need to have **at least `n_way` classes** in the dataset, because in a task you are sampling `n_way` classes from the dataset's classes. If you have less, you can't sample them.

**Comment by shraddha291996** [comment]
> You need to have **at least `n_way` classes** in the dataset, because in a task you are sampling `n_way` classes from the dataset's classes. If you have less, you can't sample them.

I was getting still getting the issue even when I had **n_way****classes**  in my dataset but when I added one extra class in my dataset the issue was resolved.

**Comment by shraddha291996** [comment]
> This shouldn't happen. Can you please reproduce the error?

Hi sorry for the confusion , after checking the test again I realized error was with number of samples in classes and not with n_ways. Thanks

**Comment by ebennequin** [comment]
Addressed with #93
